From 2d7c6370ea81cb23b8b7bc03e85b4036a236097a Mon Sep 17 00:00:00 2001
From: Alex Zhao <zzc@rock-chips.com>
Date: Thu, 14 May 2020 11:28:45 +0800
Subject: [PATCH] net: rockchip_wlan: rtl8723cs: fix some bugs

1.enable CONFIG_TXPWR_BY_RATE_EN
2.add 80211R support
3.fix EFUSE_MAP_PATH

Signed-off-by: Alex Zhao <zzc@rock-chips.com>
Change-Id: I4bbe4f78c707639e3f89070b8d6566d2ae897c9f
---
 drivers/net/wireless/rockchip_wlan/rtl8723cs/Makefile        | 5 +++--
 .../rockchip_wlan/rtl8723cs/os_dep/linux/sdio_intf.c         | 2 +-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/net/wireless/rockchip_wlan/rtl8723cs/Makefile b/drivers/net/wireless/rockchip_wlan/rtl8723cs/Makefile
index e2838b1eec06..9450f422faad 100644
--- a/drivers/net/wireless/rockchip_wlan/rtl8723cs/Makefile
+++ b/drivers/net/wireless/rockchip_wlan/rtl8723cs/Makefile
@@ -65,7 +65,7 @@ CONFIG_EFUSE_CONFIG_FILE = y
 CONFIG_EXT_CLK = n
 CONFIG_TRAFFIC_PROTECT = n
 CONFIG_LOAD_PHY_PARA_FROM_FILE = y
-CONFIG_TXPWR_BY_RATE_EN = n
+CONFIG_TXPWR_BY_RATE_EN = y
 CONFIG_TXPWR_LIMIT_EN = n
 CONFIG_RTW_CHPLAN = 0xFF
 CONFIG_RTW_ADAPTIVITY_EN = disable
@@ -1020,7 +1020,7 @@ EXTRA_CFLAGS += -DEFUSE_MAP_PATH=\"/system/etc/wifi/wifi_efuse_8189e.map\"
 else ifeq ($(MODULE_NAME), 8723bs)
 EXTRA_CFLAGS += -DEFUSE_MAP_PATH=\"/system/etc/wifi/wifi_efuse_8723bs.map\"
 else
-EXTRA_CFLAGS += -DEFUSE_MAP_PATH=\"/system/etc/wifi/wifi_efuse_$(MODULE_NAME).map\"
+EXTRA_CFLAGS += -DEFUSE_MAP_PATH=\"/vendor/etc/firmware/wifi_efuse_$(MODULE_NAME).map\"
 endif
 
 #WIFIMAC_PATH
@@ -1607,6 +1607,7 @@ EXTRA_CFLAGS += -DCONFIG_LITTLE_ENDIAN -DCONFIG_PLATFORM_ANDROID -DCONFIG_PLATFO
 # default setting for Android 4.1, 4.2, 4.3, 4.4
 EXTRA_CFLAGS += -DCONFIG_IOCTL_CFG80211 -DRTW_USE_CFG80211_STA_EVENT
 EXTRA_CFLAGS += -DCONFIG_CONCURRENT_MODE
+EXTRA_CFLAGS += -DCONFIG_RTW_80211R
 # default setting for Power control
 #EXTRA_CFLAGS += -DRTW_ENABLE_WIFI_CONTROL_FUNC
 EXTRA_CFLAGS += -DRTW_SUPPORT_PLATFORM_SHUTDOWN
diff --git a/drivers/net/wireless/rockchip_wlan/rtl8723cs/os_dep/linux/sdio_intf.c b/drivers/net/wireless/rockchip_wlan/rtl8723cs/os_dep/linux/sdio_intf.c
index 689086e6e6c0..06eab333fbfd 100644
--- a/drivers/net/wireless/rockchip_wlan/rtl8723cs/os_dep/linux/sdio_intf.c
+++ b/drivers/net/wireless/rockchip_wlan/rtl8723cs/os_dep/linux/sdio_intf.c
@@ -1121,7 +1121,7 @@ static int rtw_sdio_resume(struct device *dev)
 
 	pdbgpriv->dbg_resume_cnt++;
 
-#ifdef CONFIG_PLATFORM_INTEL_BYT
+#ifdef CONFIG_PLATFORM_ROCKCHIPS
 	if (0)
 #else
 	if (pwrpriv->wowlan_mode || pwrpriv->wowlan_ap_mode)
-- 
2.35.3

